lib/src/model/canonicalization.dart

```diff
@@ -58,7 +58,7 @@ abstract mixin class Canonicalization
     }
 
     // Give a big boost if the library has the package name embedded in it.
-    if (library.package.namePieces.intersection(library.namePieces).isEmpty) {
+    if (library.package.namePieces.intersection(library.namePieces).isNotEmpty) {
```

Comment:
It is curious that there are no test changes.

Reply:
Oof I agree. There is minimal testing of this whole method. I will add a TODO and look into adding some unit tests.